src/gui/src/PatternEditor/PatternEditorPanel.cpp

```diff
@@ -77,31 +78,31 @@ PatternEditorPanel::PatternEditorPanel( QWidget *pParent )
 	m_bCursorHidden = true;
 
 // Editor TOP
-	PixmapWidget *editor_top = new PixmapWidget(nullptr);
-	editor_top->setPixmap("/patternEditor/editor_top.png", true);
+	PixmapWidget *editor_top = new PixmapWidget( nullptr );
+	editor_top->setPixmap( "/patternEditor/editor_top.png", true );
```

Comment:
Arrgh, I am avoiding this next time.

Reply:
> Ok, could you have a look if the following is a proper use of QStrings methods for the LCD? Thanks
> ```
> 		//sprintf( tmp, "%.3f/%d", (nPatternSize * nDen)/ (float) MAX_NOTES , nDen );
> 		qtmp.setNum( ( nPatternSize * nDen ) / (float) MAX_NOTES, 'f', 3 ) += QString( "/%1" ).arg( nDen );
> ```

Looks fine, although there's no need to avoid intermediate objects (and using a function return as an lvalue looks weird), so you could write that as just:

```
qtmp = QString( "%1/%2" ).arg( ( nPatternSize * nDen ) / (float) MAX_NOTES, 'f', 3 ).arg( nDen );
```

When you do it this way, it will format using "." as the decimal separator regardless of locale (in which case `QString::toDouble` in the parsing code will be fine). If you want to use locale-specific decimal separators (I don't know if that's desirable here, you're the expert I think! :) ) then both this and the parsing code should use the QLocale versions (`QLocale::toString` and `QLocale::toDouble`).

That would make the above something like:


```
QLocale loc = QLocale::system();
qtmp = QString( "%1/%2" ).arg( loc.toString( ( nPatternSize * nDen ) / (float) MAX_NOTES, 'f', 3 ) ).arg( nDen );
```

Reply:
Thanks you for the help.
That call `arg(float, char, int)` was ambiguous but I found out the missing argument
`		qtmp =  QString( "%1/%2" ).arg( ( nPatternSize * nDen ) / (float) MAX_NOTES, 0, 'f', 3 ).arg( nDen );`

>  If you want to use locale-specific decimal separators (I don't know if that's desirable here, you're the expert I think! :) ) then both this and the parsing code should use the QLocale versions

I don't know if I deserve this choice ;)
Personally I would go with QString but all LCDs are printed in locale right now (except the Bpm LCD which has a point, but its input dialog uses comma) and I saw `sprintf()` is mainly used to set LCD tmp strings.
So better if you or the team decide :)

Reply:
Hmm, is there anywhere else in the UI where a real number can be entered by the user? I can't think of any off the top of my head but if there are it would be good to be consistent with that.

Reply:
Yes in Bpm dialog.
In Ardour both comma and point are accepted, while the display shows the value with locale separator. I think this is the behaviour we desire, consistent with the rest of the GUI.
I am doing it with`QString::replace(",", ".")` and `Qstring.ToDouble()` in the parser, and `QLocale` in LCD